app/controllers/sites/visual_designs_controller.rb

```diff
@@ -14,14 +16,30 @@ def update
     end
   end
 
+  def new_link
+    @index = params[:index].to_i
+    @type = params[:type]
+    respond_to { |format| format.js }
+  end
+
   private
 
+  def build_links
+    @site.primary_header_links = [{}] if @site.primary_header_links.blank?
```

Comment:
Unless somehow `primary_header_links` ends as an empty string we may be able to do this in here 
```ruby
@site.primary_header_links ||= [{}]
```